core/src/main/kotlin/net/corda/core/crypto/CompositeSignature.kt

```
@@ -0,0 +1,70 @@
+package net.corda.core.crypto
+
+import net.corda.core.serialization.deserialize
+import java.io.ByteArrayOutputStream
+import java.security.PrivateKey
+import java.security.PublicKey
+import java.security.Signature
+import java.security.SignatureException
+import java.security.spec.AlgorithmParameterSpec
+
+/**
+ * Dedicated class for storing a set of signatures that comprise [CompositeKey].
+ */
+class CompositeSignature : Signature(ALGORITHM) {
+    companion object {
+        val ALGORITHM = "X-Corda-CompositeSig"
+    }
+    private var buffer: ByteArrayOutputStream = ByteArrayOutputStream(1024)
+    private var verifyKey: CompositeKey? = null
+
+    override fun engineGetParameter(param: String?): Any {
+        throw UnsupportedOperationException("Composite signatures do not support any parameters")
+    }
+
+    override fun engineInitSign(privateKey: PrivateKey?) {
+        throw UnsupportedOperationException("Composite signatures must be assembled independently from signatures provided by the component private keys")
+    }
+
+    override fun engineInitVerify(publicKey: PublicKey?) {
+        if (publicKey is CompositeKey) {
+            buffer = ByteArrayOutputStream(1024)
+            verifyKey = publicKey
+        } else {
+            throw IllegalArgumentException("Key to verify must be a composite key")
```

Should throw InvalidKeyException